Add needsinfo flags

willkg opened the pull request:
r?

willkg committed:
Add needsinfo flags

Fixes #54

mythmon commented on edwin/client/static/js/components/Timeline.js:
```diff
@@ -334,6 +335,25 @@ WhiteboardGroup.propTypes = {
+        {this.props.data
+          .map(flag => (
+            <span className="NeedinfoGroup__Data">
+              ni? {flag.get('name', '')}
```
Is there a case that the flag wouldn't have a name property?

mythmon replied:
To expand, the second parameter to `Immutable.Map::get` is the default value, and is optional. If it isn't provided, `undefined` will be returned, which in this case will have the same effect. Also, I think the code you have in the store guarantees there will always be a name property (even if it is `Unknown`).

mythmon commented on edwin/client/static/js/stores/BugStore.js:
```diff
@@ -102,6 +114,9 @@ function augmentBug(bug) {
   bug = bug.set('whiteboardParsed', Immutable.fromJS(whiteboardData.parse(bug.get('whiteboard', ''))));
 
+  // Get needinfo flags for this bug.
+  bug = bug.set('needinfo', Immutable.fromJS(getNeedinfo(bug)));
```
`getNeedInfo` calls `.toJS()`, but then you immediatley re-wrap it `Immutable.fromJS`. It would probably better to remove both calls.

mythmon commented on edwin/client/static/js/stores/BugStore.js:
```diff
@@ -88,6 +88,18 @@ function getBugState(bug) {
+  // Filters the flags for needinfo flags.
+  return bug.get('flags', [])
+    .filter(flag => flag.get('name', '') === 'needinfo')
+    .map(function (flag) {
```
This should probably use a es6-style function

``` js
.map(flag => ({name: flag.get('requestee', 'Unknown'}))
```

or a slightly less terse version

``` js
.map(flag => {
  return {
    name: flag.get('requestee', 'Unknown'),
  };
})
```

willkg commented:
^^^ That addresses all the feedback plus nixes the domain name for requestee names and cleans up some whitespace issues.

PLUS ROUNDED CORNERS!!!!!11!!!

mythmon commented:
I have opinions about the styling of this, but the data half is spot on. Thanks!
